framework/auth/tasks.py

```diff
@@ -0,0 +1,16 @@
+from framework.celery_tasks import app
+
+@app.task
+def update_user_from_login(user_id, login_time, updates=None):
+    from osf.models import OSFUser
+    if not updates:
+        updates = {}
+    user = OSFUser.load(user_id)
+    user.update_date_last_login(login_time)
```

Comment:
Why not throw `login_time` into the `updates`?

Reply:
No particular reason, outside of consistency with how `date_last_login` is being updated everywhere else, and having it be a required arg to avoid having to do something like either `user.update_date_last_login(login_time=updates.get('date_last_login', None)` or
```python
if 'date_last_login' in updates:
    user.update_date_last_login(login_time=updates['date_last_login'])
else:
    user.update_date_last_login()
```